examples/specs/layer_line_multi_value_tooltip.vl.json

```diff
@@ -0,0 +1,127 @@
+{
+  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
+  "data": {
+    "url": "data/stocks.csv",
+    "format": {
+      "type": "csv"
+    }
+  },
+  "width": 800,
+  "height": 400,
+  "layer": [
+    {
+      "mark": "line",
+      "encoding": {
+        "color": {
+          "type": "nominal",
+          "field": "symbol"
+        },
+        "x": {
+          "type": "temporal",
+          "field": "date"
+        },
+        "y": {
+          "type": "quantitative",
+          "field": "price"
+        }
+      }
+    },
+    {
+      "mark": "point",
+      "encoding": {
+        "opacity": {
+          "value": 0
+        },
+        "x": {
+          "type": "temporal",
+          "field": "date"
+        }
+      },
+      "selection": {
+        "tooltip": {
+          "type": "single",
+          "nearest": true,
+          "on": "mouseover",
+          "encodings": [
+            "x"
+          ],
+          "empty": "none",
+          "resolve": "global"
+        }
+      }
+    },
+    {
+      "mark": "point",
+      "encoding": {
+        "color": {
+          "type": "nominal",
+          "field": "symbol"
+        },
+        "opacity": {
+          "condition": {
+            "selection": "tooltip",
+            "value": 1
+          },
+          "value": 0
+        },
+        "x": {
+          "type": "temporal",
+          "field": "date"
+        },
+        "y": {
+          "type": "quantitative",
+          "field": "price"
+        }
+      }
+    },
+    {
+      "mark": {
+        "type": "rule",
+        "color": "gray"
+      },
+      "encoding": {
+        "x": {
+          "type": "temporal",
+          "field": "date"
+        }
+      },
+      "transform": [
+        {
+          "filter": {
+            "selection": "tooltip"
+          }
+        }
+      ]
+    },
+    {
+      "mark": {
+        "type": "text",
+        "align": "left",
+        "dx": 5,
+        "dy": -5
+      },
+      "encoding": {
+        "color": {
+          "type": "nominal",
+          "field": "symbol"
+        },
+        "text": {
+          "condition": {
```

Comment:
It's better to group rule and text by a nested layer and use the same filter for both of them.  This way, we don't have to apply full pass to filter / condition the whole dataset twice.  Plus, the current code still creates N-1 instances of empty text in the underlying scenegraph.

Reply:
Can you give me an example of a nested layer?

Reply:
You can place a layer as a spec inside a layer.

Reply:
I am facing some issues in grouping rule and text by a nested layer. Need some help.
```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "description": "Credits to @jakevdp. The example is forked/modified from his block.",
  "data": {
    "url": "data/stocks.csv",
    "format": {
      "type": "csv"
    }
  },
  "width": 800,
  "height": 400,
  "layer": [
    {
      "mark": "line",
      "encoding": {
        "color": {
          "type": "nominal",
          "field": "symbol"
        },
        "x": {
          "type": "temporal",
          "field": "date"
        },
        "y": {
          "type": "quantitative",
          "field": "price"
        }
      }
    },
    {
      "mark": "point",
      "encoding": {
        "color": {
          "type": "nominal",
          "field": "symbol"
        },
        "opacity": {
          "value": 0,
          "condition": {
            "selection": "tooltip",
            "value": 1
          }
        },
        "x": {
          "type": "temporal",
          "field": "date"
        },
        "y": {
          "type": "quantitative",
          "field": "price"
        }
      },
      "selection": {
        "tooltip": {
          "type": "single",
          "nearest": true,
          "on": "mouseover",
          "encodings": [
            "x"
          ],
          "empty": "none",
          "resolve": "global"
        }
      }
    },
    {
      "layer": [
        {
         "mark": {
           "type": "rule",
           "color": "gray"
         },
         "encoding": {
           "x": {
             "type": "temporal",
             "field": "date"
           }
         },
         "transform": [
           {
             "filter": {
               "selection": "tooltip"
             }
           }
         ]
       },
       {
         "mark": {
           "type": "text",
           "align": "left",
           "dx": 5,
           "dy": -5
         },
         "encoding": {
           "color": {
             "type": "nominal",
             "field": "symbol"
           },
           "text": {
             "condition": {
               "type": "quantitative",
               "field": "price",
               "selection": "tooltip"
             },
             "value": " "
           },
           "x": {
             "type": "temporal",
             "field": "date"
           },
           "y": {
             "type": "quantitative",
             "field": "price"
           }
         }
       }
     ]
    }
  ]
}
```